packages/opentelemetry-plugin-winston/src/winston.ts

```diff
@@ -0,0 +1,121 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as semver from 'semver';
+import * as shimmer from 'shimmer';
+import * as winston from 'winston';
+import { BasePlugin } from '@opentelemetry/core';
+import { WinstonChunk } from './types';
+import { addTraceToChunk, processArgs } from './utils';
+import { VERSION } from './version';
+
+/**
+ * Winston instrumentation plugin for Open Telemetry
+ */
+export class WinstonPlugin extends BasePlugin<any> {
+  readonly component: string;
+
+  constructor(readonly moduleName: string, readonly version: string) {
+    super(`@opentelemetry/plugin-${moduleName}`, version);
+    this.component = this.moduleName;
+    this._config = {};
+  }
+
+  /**
+   * Patching createLogger function for winston version 3
+   */
+  private _patchCreateLoggerV3() {
+    return (
+      original: any
+    ): ((this: winston.Logger, ...args: any[]) => winston.Logger) => {
+      const plugin = this;
+      return function createLoggerPatched(
+        this: winston.Logger,
+        ...args: any[]
+      ): winston.Logger {
+        const logger: winston.Logger = original.apply(this, args);
+        const originalWrite = logger.write;
+
+        logger.write = function writePatched(this, ...args: any) {
+          const span = plugin._tracer.getCurrentSpan();
+          if (span) {
+            const chunk: WinstonChunk = args[0];
+            args[0] = addTraceToChunk(chunk, span);
+          }
+          return originalWrite.apply(logger, args);
+        };
+
+        return logger;
+      };
+    };
+  }
+
+  /**
+   * Patching log function for winston version 1 & 2
+   */
+  private _patchLogV2() {
+    return (original: any): ((this: any, ...args: any[]) => any) => {
+      const plugin = this;
+      return function logPatched(this: any, ...args: any[]): any {
+        const span = plugin._tracer.getCurrentSpan();
+        if (span) {
+          args = processArgs(args, span);
+        }
+        return original.apply(this, args);
+      };
+    };
+  }
+
+  /**
+   * implements patch function
+   */
+  protected patch() {
+    if (semver.satisfies(this._moduleExports.version, '>=3.0.0')) {
+      shimmer.wrap(
+        this._moduleExports,
+        'createLogger',
+        this._patchCreateLoggerV3()
+      );
+    }
+    if (semver.satisfies(this._moduleExports.version, '1 - 2')) {
+      shimmer.wrap(
+        this._moduleExports.Logger.prototype,
+        'log',
+        this._patchLogV2()
+      );
+    }
+    return this._moduleExports;
+  }
+
+  /**
+   * implements unpatch function
+   */
+  protected unpatch(): void {
```

Comment:
why do you think so ?

Reply:
Because shimmer is only unpatching the `createLogger` method here. Any loggers that have already been created before you unpatch have your patched `write` on them already and that isn't removed.

Reply:
fixed